webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -453,7 +453,17 @@ var KM_ATTACHMENT_V2_SUPPORTED_MIME_TYPES = ["application","text","image"];
         var isUserDeleted = false;
         var mckVideoCallringTone = null;
         var KM_ASK_USER_DETAILS = mckMessageService.checkArray(appOptions.askUserDetails);
-        var KM_PRELEAD_COLLECTION =mckMessageService.checkArray(appOptions.preLeadCollection) ? appOptions.preLeadCollection:[];
+        var KM_PRELEAD_COLLECTION;
+        if(appOptions.preLeadCollection){
+            KM_PRELEAD_COLLECTION = mckMessageService.checkArray(appOptions.preLeadCollection);
```

Comment:
is this code tested?
please share a sample of how this part of the flow was tested?

Reply:
By passing (preLeadCollection) parameters from the installation script we get this, Overwrite the dashboard settings
![image](https://user-images.githubusercontent.com/22856546/108674050-ad25f200-750a-11eb-9199-3c7879fdb693.png)

![image](https://user-images.githubusercontent.com/22856546/108674742-be233300-750b-11eb-9cd9-ee78195c91b3.png)

If we don't pass any (preLeadCollection) parameters from the installation script and have pre-chat-lead-collection settings enabled in the dashboard then it will show these options on the chat widget.
![image](https://user-images.githubusercontent.com/22856546/108675363-b6b05980-750c-11eb-963d-e0cd695f46e2.png)

![image](https://user-images.githubusercontent.com/22856546/108675292-9da7a880-750c-11eb-94a5-b19d11079d52.png)

Reply:
what is the return value for mckMessageService.checkArray(appOptions.preLeadCollection)?

in the earlier code, it was 
var KM_PRELEAD_COLLECTION =mckMessageService.checkArray(appOptions.preLeadCollection) ? appOptions.preLeadCollection:[]

which means the value was: appOptions.preLeadCollection
now this code final value is  mckMessageService.checkArray(appOptions.preLeadCollection)
which value is correct? appOptions.preLeadCollection or mckMessageService.checkArray(appOptions.preLeadCollection)